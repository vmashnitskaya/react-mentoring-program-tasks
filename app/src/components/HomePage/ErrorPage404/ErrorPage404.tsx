import React, { FunctionComponent } from 'react';
import './ErrorPage404.scss';
import { Link } from "react-router-dom";
import Button from '../../common/Button/Button';


const ErrorPage404: FunctionComponent = () => {
        return <div className="page-not-found wrapper">
            <img src="/images/404.svg" alt="404 error" />
            <Link to={"/home"}>
                <Button text="Go back home" />
            </Link>
        </div>
};

export default ErrorPage404;
