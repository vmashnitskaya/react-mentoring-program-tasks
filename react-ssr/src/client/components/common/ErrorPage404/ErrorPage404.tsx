import React, { FunctionComponent } from 'react';
import './ErrorPage404.scss';
import Button from '../Button/Button';


const ErrorPage404: FunctionComponent = () => {
        return <div className="page-not-found wrapper">
            <img src="/images/404.svg" alt="404 error" />
            <a href="/">
                <Button text="Go back home" />
            </a>
        </div>
};

export default ErrorPage404;
