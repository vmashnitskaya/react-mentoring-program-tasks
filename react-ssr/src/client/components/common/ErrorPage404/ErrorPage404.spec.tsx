import React from 'react';
import { render } from '@testing-library/react';
import ErrorPage404 from "./ErrorPage404";

describe('ErrorPage404', () =>{
    test('renders error page 404 snapshot', () => {
        const {asFragment} = render(<ErrorPage404 />);

        expect(asFragment()).toMatchSnapshot();
    });
});
