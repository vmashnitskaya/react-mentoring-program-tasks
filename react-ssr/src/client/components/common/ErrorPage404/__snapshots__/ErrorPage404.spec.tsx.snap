// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage404 renders error page 404 snapshot 1`] = `
<DocumentFragment>
  <div
    class="page-not-found wrapper"
  >
    <img
      alt="404 error"
      src="/images/404.svg"
    />
    <a
      href="/home"
    >
      <button
        class="uppercase"
        type="button"
      >
        Go back home
      </button>
    </a>
  </div>
</DocumentFragment>
`;
